rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -1291,21 +1291,43 @@ def _log_metric_info(self) -> None:
             logger.debug(f"  {metric} ({name})")
 
     def _prepare_layers(self) -> None:
+        # For user text, prepare layers that combine different feature types, embed
+        # everything using a transformer and optionally also do masked language modeling.
         self.text_name = TEXT
-        self._prepare_sequence_layers(self.text_name)
+        self._tf_layers[
+            f"{self.text_name}_sequence_layer"
+        ] = rasa_layers.RasaSequenceLayer(
+            self.text_name, self.data_signature[self.text_name], self.config
+        )
         if self.config[MASKED_LM]:
-            self._prepare_mask_lm_layers(self.text_name)
+            self._prepare_mask_lm_loss(self.text_name)
+
+        # Intent labels are treated similarly to user text but without the transformer
+        # and masked language modelling.
         if self.config[INTENT_CLASSIFICATION]:
             self.label_name = TEXT if self.config[SHARE_HIDDEN_LAYERS] else LABEL
-            self._prepare_input_layers(self.label_name)
+            self._tf_layers[
+                f"{self.label_name}_input_layer"
+            ] = rasa_layers.RasaInputLayer(
+                self.label_name, self.label_signature[self.label_name], self.config
+            )
+
+            self._prepare_ffnn_layer(
+                self.label_name,
+                self.config[HIDDEN_LAYERS_SIZES][self.label_name],
+                self.config[DROP_RATE],
+            )
+
             self._prepare_label_classification_layers()
+
         if self.config[ENTITY_RECOGNITION]:
             self._prepare_entity_recognition_layers()
 
-    def _prepare_mask_lm_layers(self, name: Text) -> None:
-        self._tf_layers[f"{name}_input_mask"] = layers.InputMask()
```

Comment:
Because it's done in `RasaSequenceLayer` which prepares this mask and also the token IDs for MLM. Outside of that layer you just embed everything and pass it to the loss function.